*: Generate all manifests

brancz opened the pull request:
**What this PR does / why we need it**:

Sharding and autosharding introduced a number of manifests that are
very similar, but not identical to the standard manifests. This
introduces generating all manifests using jsonnet and moves them to
the explicit `/examples` directory.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #903

LiliC commented on README.md:
```diff
@@ -214,7 +214,7 @@ service account token that has read-only access to the Kubernetes cluster.
 #### Kubernetes Deployment
 
-To deploy this project, you can simply run `kubectl apply -f kubernetes` and a
+To deploy this project, you can simply run `kubectl apply -f kubernetes/standard` and a
```
standard? Hmm we can come up with a better name I think here.

brancz replied:
Any suggestions? I haven't been able to come up with anything better, and this is the standard way to deploy kube-state-metrics.

LiliC replied:
We said we want to also since we moving it, rename the directory to `examples`.

LiliC commented on kubernetes/autosharding/service.yaml:
```diff
@@ -1,21 +1,19 @@
 metadata:
+  labels:
+    app.kubernetes.io/name: kube-state-metrics
+    app.kubernetes.io/version: v1.8.0-rc.0
```
Should not all be `v1.8.0-rc.1`?

brancz replied:
It's whatever is in the `VERSION` file

LiliC replied:
Not true, in the file above its v1.8.0-rc.1.

LiliC commented on kubernetes/autosharding/statefulset.yaml:
```diff
@@ -1,53 +1,58 @@
 metadata:
+  labels:
+    app.kubernetes.io/name: kube-state-metrics
+    app.kubernetes.io/version: v1.8.0-rc.0
```
v1.8.0-rc.1?

LiliC commented on scripts/jsonnetfile.json:
```diff
@@ -0,0 +1,13 @@
+            "name": "kube-state-metrics",
+            "source": {
+                "local": {
+                    "directory": "../jsonnet/kube-state-metrics"
```
Where is this directory? 🤔

brancz replied:
`/jsonnet` is in the root of the repo

LiliC commented on scripts/jsonnetfile.json:
```diff
@@ -0,0 +1,13 @@
+                    "directory": "../jsonnet/kube-state-metrics"
+                }
+            },
+            "version": ""
```
Should we pin this to master?

brancz replied:
it's a directory based package so it will always do whatever is currently in the directory

brancz commented:
Added a note to the release doc, and I believe I addressed all comments, if you have a suggestion for the name of the "standard" deployment, then I'm happy to adapt it.

LiliC commented on kustomization.yaml:
```diff
@@ -1,11 +1,8 @@
-- kubernetes/kube-state-metrics-deployment.yaml
-- kubernetes/kube-state-metrics-service-account.yaml
-- kubernetes/kube-state-metrics-service.yaml
+- kubernetes/cluster-role-binding.yaml
```
Should this be examples?

brancz replied:
good catch, forgot about this one

LiliC commented on tests/e2e.sh:
```diff
@@ -117,20 +117,20 @@ KUBE_STATE_METRICS_IMAGE_TAG=$(docker images -a|grep 'quay.io/coreos/kube-state-
 
 # set up kube-state-metrics manifests
-kubectl create -f ./kubernetes/kube-state-metrics-service-account.yaml
+kubectl create -f ./kubernetes/standard/service-account.yaml
```
This should be examples now here as well.
